File: 351004/Barsukov/app/services/sticker_service.py
from sqlalchemy.orm import Session
from repository import StickerRepo
from app.schemas.sticker import StickerRequestTo

class StickerService:
    def __init__(self):
        self.repo = StickerRepo()

    def create(self, db: Session, dto: StickerRequestTo):
        return self.repo.create(db, dto.model_dump(exclude_none=True))

    def get_all(self, db: Session):
        return self.repo.get_all(db)

    def get_by_id(self, db: Session, id: int):
        return self.repo.get_by_id(db, id)

    def update(self, db: Session, id: int, dto: StickerRequestTo):
        return self.repo.update(db, id, dto.model_dump(exclude_none=True))

    def delete(self, db: Session, id: int):
        return self.repo.delete(db, id)

File: 351004/Levental/task310-rest/src/main/java/by/bsuir/task310/service/StickerService.java
package by.bsuir.task310.service;

import by.bsuir.task310.domain.Sticker;
import by.bsuir.task310.dto.request.StickerRequestTo;
import by.bsuir.task310.dto.response.StickerResponseTo;
import by.bsuir.task310.mapper.StickerMapper;
import by.bsuir.task310.repository.StickerRepository;
import by.bsuir.task310.web.error.NotFoundException;
import by.bsuir.task310.web.error.ValidationException;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class StickerService {

    private final StickerRepository repo;
    private final StickerMapper mapper;

    public StickerService(StickerRepository repo, StickerMapper mapper) {
        this.repo = repo;
        this.mapper = mapper;
    }

    public StickerResponseTo create(StickerRequestTo dto) {
        Sticker entity = mapper.toEntity(dto);
        Sticker saved = repo.save(entity);
        return mapper.toResponse(saved);
    }

    public List<StickerResponseTo> getAll() {
        return repo.findAll().stream()
                .map(mapper::toResponse)
                .toList();
    }

    public StickerResponseTo getById(Long id) {
        Sticker entity = repo.findById(id)
                .orElseThrow(() -> new NotFoundException("sticker not found", "01"));
        return mapper.toResponse(entity);
    }

    public StickerResponseTo update(StickerRequestTo dto) {
        if (dto.getId() == null) {
            throw new ValidationException("id is required", "01");
        }

        if (!repo.existsById(dto.getId())) {
            throw new NotFoundException("sticker not found", "02");
        }

        Sticker entity = mapper.toEntity(dto);
        Sticker updated = repo.update(dto.getId(), entity);
        return mapper.toResponse(updated);
    }

    public void delete(Long id) {
        if (!repo.existsById(id)) {
            throw new NotFoundException("sticker not found", "03");
        }
        repo.delete(id);
    }
}